shm/Game.cpp

```diff
@@ -86,6 +84,8 @@ void Game::travel()
     else
     {
         std::cout << "There is no such island on the map! Choose another destination!" << '\n';
+        std::cin.clear();
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
```

Comment:
This is repeated few times, maybe create function `clearInputStream()`? 
DRY rule -> don't repeat yourself :)